src/QuickBooks/Provider.php

```diff
@@ -2,25 +2,87 @@
 
 namespace SocialiteProviders\QuickBooks;
 
-use SocialiteProviders\Manager\OAuth1\AbstractProvider;
-use SocialiteProviders\Manager\OAuth1\User;
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+use Laravel\Socialite\Two\ProviderInterface;
+use SocialiteProviders\Manager\ConfigTrait;
 
-class Provider extends AbstractProvider
+class Provider extends AbstractProvider implements ProviderInterface
 {
+    use ConfigTrait;
+
     /**
      * Unique Provider Identifier.
      */
     const IDENTIFIER = 'QUICKBOOKS';
 
+    /**
+     * Sandbox endpoint for retrieving user info.
+     */
+    const USERSANDBOXENDPOINT = 'https://sandbox-accounts.platform.intuit.com/v1/openid_connect/userinfo';
+
+    /**
+     * Production endpoint for retrieving user info.
+     */
+    const USERPRODUCTIONENDPOINT = 'https://accounts.platform.intuit.com/v1/openid_connect/userinfo';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopes = ['com.intuit.quickbooks.accounting openid profile email phone address'];
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        return $this->buildAuthUrlFromBase('https://appcenter.intuit.com/connect/oauth2', $state);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenUrl()
+    {
+        return 'https://oauth.platform.intuit.com/oauth2/v1/tokens/bearer';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getUserByToken($token)
+    {
+        $endpoint = $this->getConfig('app.env', 'local') == 'production' ? self::USERPRODUCTIONENDPOINT : self::USERSANDBOXENDPOINT;
```

Comment:
can you split this onto multiple lines please?

Reply:
Also, strict comparison please `===`.
I would also prefer to make this configurable rather than using the current env?

Reply:
good idea - added 👍